src/pages/profile/components/Sidebar.tsx

```diff
@@ -140,14 +156,35 @@ export default function Sidebar({
     Object.keys(updatedData).forEach((key: string) => {
       const typedKey = key as keyof User
 
-      if (!isInvalidInput(updatedData[typedKey], 'string', true) && originalData[typedKey] !== updatedData[typedKey]) {
+      if (
+        !isInvalidInput(updatedData[typedKey], ['string', 'boolean'], true) &&
+        originalData[typedKey] !== updatedData[typedKey]
+      ) {
         changes[typedKey] = updatedData[typedKey]
       }
     })
 
     return changes
   }
 
+  function onUsernameChange(value: string, type?: string) {
```

Comment:
`isUsernameArNS` is undefined and when i try to update custom username, the value of it is not updated here and `isUsernameArNS` is required in the schema so on clicking `Save Details` the form is not submitted.